File: software/examples/perl/example_threshold.pl
#!/usr/bin/perl

use strict;
use Tinkerforge::IPConnection;
use Tinkerforge::BrickletHumidityV2;

use constant HOST => 'localhost';
use constant PORT => 4223;
use constant UID => 'XYZ'; # Change XYZ to the UID of your Humidity Bricklet 2.0

# Callback subroutine for humidity callback
sub cb_humidity
{
    my ($humidity) = @_;

    print "Humidity: " . $humidity/100.0 . " %RH\n";
    print "Recommended humiditiy for human comfort is 30 to 60 %RH.\n";
}

my $ipcon = Tinkerforge::IPConnection->new(); # Create IP connection
my $h = Tinkerforge::BrickletHumidityV2->new(&UID, $ipcon); # Create device object

$ipcon->connect(&HOST, &PORT); # Connect to brickd
# Don't use device before ipcon is connected

# Register humidity callback to subroutine cb_humidity
$h->register_callback($h->CALLBACK_HUMIDITY, 'cb_humidity');

# Configure threshold for humidity "outside of 30 to 60 %RH"
# with a debounce period of 10s (10000ms)
$h->set_humidity_callback_configuration(10000, 0, 'o', 30*100, 60*100);

print "Press key to exit\n";
<STDIN>;
$ipcon->disconnect();

File: software/examples/javascript/ExampleThreshold.js
var Tinkerforge = require('tinkerforge');

var HOST = 'localhost';
var PORT = 4223;
var UID = 'XYZ'; // Change XYZ to the UID of your Humidity Bricklet 2.0

var ipcon = new Tinkerforge.IPConnection(); // Create IP connection
var h = new Tinkerforge.BrickletHumidityV2(UID, ipcon); // Create device object

ipcon.connect(HOST, PORT,
    function (error) {
        console.log('Error: ' + error);
    }
); // Connect to brickd
// Don't use device before ipcon is connected

ipcon.on(Tinkerforge.IPConnection.CALLBACK_CONNECTED,
    function (connectReason) {
        // Configure threshold for humidity "outside of 30 to 60 %RH"
        // with a debounce period of 10s (10000ms)
        h.setHumidityCallbackConfiguration(10000, false, 'o', 30*100, 60*100);
    }
);

// Register humidity callback
h.on(Tinkerforge.BrickletHumidityV2.CALLBACK_HUMIDITY,
    // Callback function for humidity callback
    function (humidity) {
        console.log('Humidity: ' + humidity/100.0 + ' %RH');
        console.log('Recommended humidity for human comfort is 30 to 60 %RH.');
    }
);

console.log('Press key to exit');
process.stdin.on('data',
    function (data) {
        ipcon.disconnect();
        process.exit(0);
    }
);
